src/target/ext/catch_syscalls.rs

```diff
@@ -0,0 +1,50 @@
+//! Enable or disable catching syscalls from the inferior process.
+
+use crate::arch::Arch;
+use crate::target::Target;
+
+/// Target Extension - Enable and disable catching syscalls from the inferior
+/// process.
+///
+/// Corresponds GDB's [`QCatchSyscalls`](https://sourceware.org/gdb/current/onlinedocs/gdb/General-Query-Packets.html#QCatchSyscalls) command.
+pub trait CatchSyscalls: Target {
+    /// Enables catching syscalls from the inferior process.
+    ///
+    /// If `filter` not `None`, then only the syscalls listed in the filter
+    /// should be reported to GDB.
+    ///
+    /// Note: filters are not combined, subsequent calls this method should
+    /// replace any existing syscall filtering.
+    fn enable_catch_syscalls(
+        &mut self,
+        filter: Option<SyscallNumbers<<Self::Arch as Arch>::Usize>>,
+    ) -> Result<(), Self::Error>;
```

Comment:
`QCatchSyscalls` supports returning error codes, so these methods should use `TargetResult`.

As a more general point, in an ideal world, _all_ GDB RSP methods would support returning errors, but thanks to our good friend "backwards compatibility", many of the older packets are stuck with having no protocol-level error handling...